### PR TITLE
Implement Outgoing transfers tab

### PR DESCRIPTION
**Pull request by sprice**
https://codexprotocol.atlassian.net/projects/SCRUM/issues/SCRUM-238

### CODE REVIEW

File: src/components/TitleTransferOutgoingListItem.vue [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    },
+    contract() {
+      return this.web3.contractInstance()
+    },
+  },
+  methods: {
+    viewTitle() {
+      this.$router.push(this.route)
+    },
+    acceptTransfer() {
```
Comment (johnhforrest) [inline src/components/TitleTransferOutgoingListItem.vue]: bunch of this is incoming specific

File: src/views/TransferIncomingListView.vue [hunk 2]
```diff
@@ -3,9 +3,9 @@
     <app-header title="Transfers" />
     <div class="sub-heading">
```
Comment (johnhforrest) [inline src/views/TransferIncomingListView.vue]: I think we should break this subheader out into it's own component and just pass some property that indicates which tab is currently active

Comment (johnhforrest) [reply src/views/TransferIncomingListView.vue]: (That way you don't have to maintain this active/unactive state on every settings page)

File: src/views/TransferOutgoingListView.vue [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+export default {
+  name: 'title-list',
+  components: {
+    AppHeader,
+    TitleTransferOutgoingListItem,
+  },
+  data() {
+    return {
+      titles: [],
+      columnNames: ['Preview', 'Piece title', 'Created at'],
```
Comment (johnhforrest) [inline src/views/TransferOutgoingListView.vue]: Don't think these are used anywhere

File: src/views/TransferOutgoingListView.vue [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        }
+      }).catch((error) => {
+        console.log('there was an error fetching outgoing transfers', error)
+      })
+    },
+  },
+}
+</script>
+
+<style lang="stylus" scoped>
```
Comment (johnhforrest) [inline src/views/TransferOutgoingListView.vue]: If you end up deciding it's cumbersome to create a single component that handles all of the various cards/lists, we should at least leverage some of Stylus' ability to reduce duplication (e.g., break this out into it's own file and import when needed)

File: src/components/TitleTransferOutgoingListItem.vue [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      const input = [
+        this.codexTitle.ownerAddress,
+        this.web3.account,
+        this.codexTitle.tokenId,
+      ]
+
+      callContract(this.contract.safeTransferFrom, input, this.web3)
```
Comment (johnhforrest) [inline src/components/TitleTransferOutgoingListItem.vue]: To cancel transferring, you basically send another "approve" transaction to the 0 address

Comment (sprice) [reply src/components/TitleTransferOutgoingListItem.vue]: Manually setting the zero address leads to an error. https://github.com/MetaMask/metamask-extension/issues/3140

Manually setting a different account address works as expected.

**Comment by sprice** [comment]
My comment collapsed with a push. Trying to cancel by transferring to zero address returns an error when trying to call the contract method.

File: src/components/TitleTransferOutgoingListItem.vue [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    contract() {
+      return this.web3.contractInstance()
+    },
+  },
+  methods: {
+    viewTitle() {
+      this.$router.push(this.route)
+    },
+    cancelTransfer() {
+      const input = ['0x0000000000000000000000000000000000000000', this.codexTitle.tokenId]
```
Comment (johnhforrest) [inline src/components/TitleTransferOutgoingListItem.vue]: not for this review but let's move the 0x0 address to a constants file at some point

**Pull request merged**